target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-301w.dts

```diff
@@ -0,0 +1,836 @@
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+	
+	button_pins: button_pins {
+		wps_button {
+			pins = "gpio57";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+		
+		rst_button {
+			pins = "gpio67";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-up;
+		}; 
+	};
+
+	/*
+	 *remove qpic pins due to device is using all of the pins for leds-gpio
+	 */
+	qpic_pins: qpic-pins {
+			/delete-property/ pins;
+			};
+
+	leds_pins: leds_pinmux {
+		led_pwr_green {
+			pins = "gpio4";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_system_green {
+			pins = "gpio1";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_system_red {
+			pins = "gpio3";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan1_green {
+			pins = "gpio6";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan1_amber {
+			pins = "gpio7";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};  
+		led_lan2_green {
+			pins = "gpio8";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan2_amber {
+			pins = "gpio11";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan3_green {
+			pins = "gpio12";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan3_amber {
+			pins = "gpio13";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_lan4_green {
+			pins = "gpio14";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan4_amber {
+			pins = "gpio15";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_10g_1_green {
+			pins = "gpio51";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_10g_1_amber {
+			pins = "gpio52";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_10g_2_green {
+			pins = "gpio54";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_10g_2_amber {
+			pins = "gpio56";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_wifi_green {
+			pins = "gpio42";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+    };
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&prng {
+	status = "okay";
+};
+
+&ssphy_0 {
+	status = "ok";
+};
+
+&qusb_phy_0 {
+	status = "ok";
+};
+
+&ssphy_1 {
+	status = "ok";
+};
+
+&qusb_phy_1 {
+	status = "ok";
+};
+
+&usb_0 {
+	status = "ok";
+};
+
+&usb_1 {
+	status = "ok";
+};
+
+
+&cryptobam {
+	status = "okay";
+};
+
+&crypto {
+	status = "okay";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qpic_nand {
+	status = "disabled";
+
+};
+
+&blsp1_spi1 { /* BLSP1 QUP1 */
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	cs-select = <0>;
```

Comment:
This doesn't exist, use cs-gpios instead

Reply:
sorry, can you give me a hint? I can't find cs-gpios in the ipq8074.dtsi either.
I would just remove that property.

Reply:
You cant remove it, it's required.
You won't find it in the DTSI as its device-specific, here is the binding:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/devicetree/bindings/spi/spi-controller.yaml?h=v5.15.1#n28

In your case `cs-gpios = <0>;` should be what you are looking for as that just means to use the HW CS

Reply:
thanks for the explanation, added.